### PR TITLE
Automated integration tests

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/106

Todo:
- [x] Test out Travis build matrix
- [x] Set up EC2 box, test script
- [x] Add new build script to launch either unit or integration tests
- [x] Rebuild SSH key and server
- [x] Document server set up
- [x] Clean up fabric output
- [x] Fix integration tests
- [x] Make integration tests not interfere with each other on the same cluster / cleanup after themselves - remove global fleet unit destruction
- [x] Look into encrypting SSH key
- [x] Add regenerated SSH key to keepass
- [x] Put in ticket for launching new machines on each integration test

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
I've removed the update integration test as I was having serious issues getting it to work. Secondly, although the int test is green, there's definite flapping.

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

**Comment by JosephSalisbury** [comment]
Note: I have limited travis to 1 concurrent job

File: .travis.yml [hunk 1]
```diff
@@ -6,29 +6,28 @@ go:
...
+        - ./builder-cache
 
 notifications:
     email: false
 
 before_install:
-- cp ./builder-cache/builder-$BUILDER_VERSION ./builder || wget https://downloads.giantswarm.io/builder/$BUILDER_VERSION/builder && chmod +x ./builder
-- export PATH=$PATH:$PWD
-- go get -u github.com/golang/lint/golint
+    - openssl aes-256-cbc -K $encrypted_314659c9f31c_key -iv $encrypted_314659c9f31c_iv -in inago-integration-test.pem.enc -out ./inago-integration-test.pem -d
```
Comment (zeisss) [inline .travis.yml]: can you add some comments what you do here and maybe add a simple existence test before?

```
test -z "encrypted_314659c9f31c_key" && echo "No \$encrypted_314659c9f31c_key given. Mandatory." && exit 1
```

or sthg

File: .travis.yml [hunk 2]
```diff
@@ -6,29 +6,28 @@ go:
...
+    - chmod +x ./builder
+    - export PATH=$PATH:$PWD
+    - go get -u github.com/golang/lint/golint
 
 script:
-- make lint
-- make ci-test
-- make ci-build
-- ./inagoctl
+    - ./ci.sh
```
Comment (zeisss) [inline .travis.yml:33]: didn't you opt against this in the past? Whats the reason for doing this with the `matrix` option above now?

File: ci.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    bash <(curl -s https://codecov.io/bash)
+elif [ $TEST_SUITE == "integration" ]; then
+    pip install --user fabric
+    make ci-build
+    eval "$(ssh-agent -s)"
+    chmod 400 ./inago-integration-test.pem
+    ssh-add ./inago-integration-test.pem 2>/dev/null
+    fab run_int_test
+else
+    echo "Unknown test suite"
```
Comment (zeisss) [inline ci.sh:17]: print out the `$TEST_SUITE` here please

File: docs/integration-server-setup.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Integration Test Server Setup
+
+This document describes the set up for the integration test server.
```
Comment (zeisss) [inline docs/integration-server-setup.md:3]: we should automate setting that up at some point too :dancer:

Comment (JosephSalisbury) [inline docs/integration-server-setup.md:3]: https://github.com/giantswarm/inago/issues/145

File: ci.sh (hunk 3 above)
Comment (JosephSalisbury) [inline ci.sh:17]: This is already done by Travis, e.g: https://travis-ci.org/giantswarm/inago/jobs/116946691#L137

File: .travis.yml (hunk 2 above)
Comment (JosephSalisbury) [inline .travis.yml:33]: I don't love it, but we need a conditional over `TEST_SUITE`. This means that we launch two jobs, due to the matrix, and the script then decides what commands to run based on `TEST_SUITE`. Can't do that in just the script section.

File: int-tests/002-update.t [hunk 5]
```diff
@@ -1,45 +0,0 @@
-Create a group to play around with.
```
Comment (zeisss) [inline int-tests/002-update.t:1]: why was this deleted?

File: .travis.yml (hunk 1 above)
Comment (JosephSalisbury) [inline .travis.yml]: I'll add a comment. `openssl` will break if the key isn't present, so I don't see the point of the test.

File: .travis.yml (hunk 2 above)
Comment (zeisss) [inline .travis.yml:33]: And this is just so it's faster?

File: int-tests/002-update.t (hunk 5 above)
Comment (JosephSalisbury) [inline int-tests/002-update.t:1]: Because it was super borked on my branch, and it's still present in your branch. My plan was to merge this, and then merge yours.

Comment (JosephSalisbury) [inline int-tests/002-update.t:1]: Ah, but you just merged yours already.

File: .travis.yml (hunk 1 above)
Comment (zeisss) [inline .travis.yml]: Lets quote the variables then, so they get passed as an empty string argument if not present at least.

File: .travis.yml (hunk 2 above)
Comment (JosephSalisbury) [inline .travis.yml:33]: yes, this means the unit and integration tests run in parallel

File: int-tests/002-update.t (hunk 5 above)
Comment (zeisss) [inline int-tests/002-update.t:1]: can you bring that file back?

**Comment by zeisss** [comment]
Bring back the deleted test, then LGTM
